app/models/content_extraction/parsers/body_content_parser.rb

```diff
@@ -0,0 +1,34 @@
+class ContentExtraction::Parsers::BodyContentParser
+  def parse(json)
+    json.dig("details", "body")
+  end
+end
+schemas_with_body = %w[
```

Comment:
This is cool, but I would rather not having code that initializes the parsers in plain ruby code. I know this is just a personal preference, and hopefully you are ok with it. 

I would be explicit about what parsers are registered in the main Parser class. Adding new parsers is quite an exceptional situation, and having more explicit code could be benefitial. Sorry, if I had not been clear about it before.

I would move this into something like:

```ruby
class ContentExtraction::Parsers::BodyContentParser
  def parse(json)
    json.dig("details", "body")
  end

  def schemas
   # List of schemas I can parse
  end
end
```

Then, the main class just need to register the same parser for each schema. It is more rudimentary, but it feels clearer to me.

Reply:
Addressed in https://github.com/alphagov/content-performance-manager/pull/576